### PR TITLE
Update Dockerfile for better user experience

### PR DESCRIPTION
**Pull request by titom73**
Summary
-------

When using this image as devcontainer in Vscode, variables are not
passed by default and raise error.

Fix define default values for ID and GID like UNAME and UPASS. This
vlaues are still override by Makefile

Build made by vscode:
---------------------

```
[docker@aa9ba99c2420 arista-ansible]$ id
uid=1000(docker) gid=1000(docker) groups=1000(docker),1002(sudo)
[docker@aa9ba99c2420 arista-ansible]$
```

Build made by Makefile

```
[docker@6eb7a519a3e6 ansible_avd]$ id
uid=501(docker) gid=20(games) groups=20(games),1002(sudo)
[docker@6eb7a519a3e6 ansible_avd]$ exit
```

Vscode configuration:

```json
// .devcontainer/devcontainer.json
{
	"name": "Existing Dockerfile",
	"context": "..",
	"dockerFile": "../development/Dockerfile",
	"settings": {
		"terminal.integrated.shell.linux": null
	}
}
```

### CODE REVIEW

**Comment by titom73** [comment]
Have to add additional improvements: 

- [x] bashrc with custom `$PS1` with git info and folder information
- [x] Install [git-extras](https://github.com/tj/git-extras/blob/master/Commands.md) extension
- [ ] Install github CLI
- [x] Share ssh-agent socket to optionally fetch and pull from container

**Comment by titom73** [comment]
Now, default shell will look like:

image

__Current ZSH Plugins installed__

- [ansible](https://github.com/ohmyzsh/ohmyzsh/tree/master/plugins/ansible)
- [common-aliases](https://github.com/ohmyzsh/ohmyzsh/tree/master/plugins/common-aliases)
- [safe-paste](https://github.com/ohmyzsh/ohmyzsh/tree/master/plugins/safe-paste)
- [git](https://github.com/ohmyzsh/ohmyzsh/tree/master/plugins/git)
- [git-extras](https://github.com/ohmyzsh/ohmyzsh/tree/master/plugins/git-extras)
- [jsontools](https://github.com/ohmyzsh/ohmyzsh/tree/master/plugins/jsontools)
- [history](https://github.com/ohmyzsh/ohmyzsh/tree/master/plugins/history)

**Comment by titom73** [comment]
Implement ssh-agent in dockerfile and expose user's ssh folder into
container for git over ssh

Because for Macos and Win, docker use VM, [there is a limitation](https://forums.docker.com/t/can-we-re-use-the-osx-ssh-agent-socket-in-a-container/8152/4) to expose ssh-agent through docker.


Example:
--------

```
$ make run
docker run --rm -it --name ansible_avd_2.9.6 -v /Users/tgrimonet/.ssh:'/home/docker'/.ssh -v /Users/tgrimonet/Projects/arista-forked-ansible/:/ansible_avd -v /etc/hosts:/etc/hosts ansible_avd:2.9.6
Agent pid 44

➜  /ansible_avd cd ansible-avd

➜  ansible-avd git:(issues/docker-vscode) ✗ git remote -v
origin  git@github.com:titom73/ansible-avd.git (fetch)
origin  git@github.com:titom73/ansible-avd.git (push)
upstream        git@github.com:aristanetworks/ansible-avd.git (fetch)
upstream        git@github.com:aristanetworks/ansible-avd.git (push)

➜  ansible-avd git:(issues/docker-vscode) ✗ git fetch upstream
Enter passphrase for key '/home/docker/.ssh/id-titom-git':
remote: Enumerating objects: 14, done.
remote: Counting objects: 100% (14/14), done.
remote: Compressing objects: 100% (12/12), done.
remote: Total 14 (delta 1), reused 0 (delta 0), pack-reused 0
Unpacking objects: 100% (14/14), done.
From github.com:aristanetworks/ansible-avd
    867f5c3..95c75df  releases/v1.0.x -> upstream/releases/v1.0.x
    92defec..2f7fb20  releases/v1.1.x -> upstream/releases/v1.1.x
```

**Comment by titom73** [comment]
Github CLI is out of scope and is not installed since a user configuration must be present on the host to make it works.

**Comment by titom73** [comment]
- Reduce size (-46%) of generated docker image by reducing number or layer.
- Use official Red Hat method to configure Python 3 as default version.

Results:
--------

```
$ docker images | grep avd
ansible_avd         2.9.6               49e81b02f925        2 days ago          806MB
ansible_avd         original            49e81b02f925        2 days ago          806MB

$ make build

$ docker images | grep avd
ansible_avd         2.9.6               d4d950aade95        10 seconds ago      442MB
ansible_avd         original            49e81b02f925        2 days ago          806MB
```

File: development/Dockerfile [hunk 1]
```diff
@@ -1,66 +1,77 @@
...
 ADD requirements.txt /tmp/requirements.txt
 ADD requirements-dev.txt /tmp/requirements-dev.txt
-RUN pip3 install -r /tmp/requirements.txt
-RUN pip3 install -r /tmp/requirements-dev.txt
-ARG ANSIBLE=2.9.6
-RUN pip3 install ansible==$ANSIBLE
-
-# Clean up
-RUN yum clean all
+RUN pip3 install -r requirements.txt \
```
Comment (carlbuchmann) [inline development/Dockerfile]: replace with: `RUN pip3 install -r /tmp/requirements.txt \`

File: development/Dockerfile [hunk 2]
```diff
@@ -1,66 +1,77 @@
...
+   && pip3 install -r requirements-dev.txt \
```
Comment (carlbuchmann) [inline development/Dockerfile]: replace with: `   && pip3 install -r /tmp/requirements-dev.txt \`

File: development/Dockerfile (hunk 1 above)
Comment (titom73) [reply development/Dockerfile]: Code updated with commit 07dd8c1

File: development/Dockerfile (hunk 2 above)
Comment (titom73) [reply development/Dockerfile]: Code updated with commit 07dd8c1

**Review by carlbuchmann — COMMENTED** [review]
After testing I incurred some issues with pytest. I resolved by updating  to latest version.
Please update requirements.txt file with the following versions + update ward version:

```
pytest==5.4.2
pytest-html==2.1.1
ward==0.46.0b0
```

**Comment by carlbuchmann** [comment]
LGTM

**Pull request merged**